captum/attr/_core/layer/layer_feature_ablation.py

```diff
@@ -0,0 +1,270 @@
+                            samples.
+                            Default: 1
+
+            Returns:
+                *tensor* of **attributions**:
+                - **attributions** (*tensor*):
+                            Attribution of each neuron in given layer input or
+                            output. Attributions will always be the same size as
+                            the input or output of the given layer, depending on
+                            whether we attribute to the inputs or outputs
+                            of the layer which is decided by the input flag
+                            `attribute_to_layer_input`.
+
+            Examples::
+
+            >>> # SimpleClassifier takes a single input tensor of size Nx4x4,
+            >>> # and returns an Nx3 tensor of class probabilities.
+            >>> # It contains an attribute conv1, which is an instance of nn.conv2d,
+            >>> # and the output of this layer has dimensions Nx12x3x3.
+            >>> net = SimpleClassifier()
+            >>> # Generating random input with size 2 x 4 x 4
+            >>> input = torch.randn(2, 4, 4)
+            >>> # Defining LayerFeatureAblation interpreter
+            >>> ablator = LayerFeatureAblation(net, net.conv1)
+            >>> # Computes ablation attribution, ablating each of the 108
+            >>> # neurons independently.
+            >>> attr = ablator.attribute(input, target=1)
+
+            >>> # Alternatively, we may want to ablate neurons in groups, e.g.
+            >>> # grouping all the layer outputs in the same row.
+            >>> # This can be done by creating a layer mask as follows, which
+            >>> # defines the groups of layer inputs / outouts, e.g.:
+            >>> # +---+---+---+
+            >>> # | 0 | 0 | 0 |
+            >>> # +---+---+---+
+            >>> # | 1 | 1 | 1 |
+            >>> # +---+---+---+
+            >>> # | 2 | 2 | 2 |
+            >>> # +---+---+---+
+            >>> # With this mask, all the 36 neurons in a row / channel are ablated
+            >>> # simultaneously, and the attribution for each neuron in the same
+            >>> # group (0 - 2) per example are the same.
+            >>> # The attributions can be calculated as follows:
+            >>> # layer mask has dimensions 1 x 3 x 3
+            >>> layer_mask = torch.tensor([[[0,0,0],[1,1,1],
+            >>>                             [2,2,2]]])
+            >>> attr = ablator.attribute(input, target=1,
+            >>>                          layer_mask=layer_mask)
+        """
+
+        def layer_forward_func(*args):
+            layer_input = args[0]
+            original_inputs = args[1:]
+
+            device_ids = self.device_ids
+            if device_ids is None and isinstance(
+                self.forward_func, torch.nn.DataParallel
+            ):
+                device_ids = getattr(self.forward_func, "device_ids", None)
```

Comment:
I'm not able to comment in our previous conversation thread about the `device_ids ` - I'll comment here:
I think currently we have some places where we check with `torch.nn.DataParallel` and some cases where we don't (e.g in `gradient.py`).
If we don't explicitly check for `torch.nn.DataParallel` then it is more generic but the user might not always want it as you mentioned. 
But from other point of view they can always set `self.forward_func.device_ids=None` if they don't want the computations to be performed in a distributed manner ?

Reply:
Makes sense, sounds good, it would be good to choose one approach and stay consistent. Should we not check for torch.nn.DataParallel everywhere when accessing device_ids and update documentation to reflect this?

Reply:
Yeah, probably that would be good - in case we want to also support ModelParallel, DistributedDataParallel in the future and custom forward functions.

Reply:
Sounds good